src/tree/ProductionSlotTreeItem.ts

```diff
@@ -16,11 +17,17 @@ export class ProductionSlotTreeItem extends SlotTreeItemBase {
 
     private readonly _slotsTreeItem: SlotsTreeItem;
 
-    public constructor(parent: AzureParentTreeItem, client: SiteClient) {
+    private constructor(parent: FunctionAppProvider, client: SiteClient) {
         super(parent, client);
         this._slotsTreeItem = new SlotsTreeItem(this);
     }
 
+    public static async create(parent: FunctionAppProvider, client: SiteClient): Promise<ProductionSlotTreeItem> {
```

Comment:
I like this design pattern.  But is there a particular reason you're implementing this in this PR?

Reply:
Because of this change: https://github.com/Microsoft/vscode-azurefunctions/pull/1213/files#diff-c071ebec37986c057b8e2e46077ecbd7R72

I need to make sure refreshImpl happens when the tree item is first created